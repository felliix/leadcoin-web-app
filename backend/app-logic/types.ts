import UploadCSV from "./upload-csv"
import UploadForm from "./upload-form"
import UploadLeads from "./leads"
import auth from "./auth"

import EmailCreator from "../models/email-creator/email-creator"
import EmailSender from "../models/emailsender/abstraction"

export interface appModels {
  emailCreator: EmailCreator
  emailSender: EmailSender
}
export interface appLogic {
  uploadCSV: UploadCSV
  uploadForm: UploadForm
  uploadLeads: UploadLeads
  auth: auth
}
