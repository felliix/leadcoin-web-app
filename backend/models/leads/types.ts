import { Currency } from "../../utils/currency"

export interface LeadQueryOptions {
  id?: number
  condition?: any
  sort?: any
  sort_by?: any
  email?: string
  fields?: string[]
  page?: number
  limit?: any
  bought_from?: number
  ownerId?: number
  active?: boolean
  filters?: object[]
}

export interface BaseLead {
  id?: number
  Price: number
  Description: string
  "Lead Price": number
  date: number
  Telephone: string
  bought_from: number | null
  ownerId: number
  forSale: boolean
  active: boolean
  meta?: any
}

export interface NewBaseLead {
  date: number
  Telephone?: string
  "Lead Price": any
  bought_from?: number | null
  forSale?: boolean
  active: boolean
  Description?: string
  agree_to_terms?: boolean
  meta?: any
}
export interface NewRealEstateLead extends NewBaseLead {
  "Lead Type": "realestate"
  Description: string
  Price: number
  "Bedrooms/Baths": string
  "Contact Person": string
  Type: "Sell" | "Rent"
  Size?: number
  State?: string
  Location?: number
  "Housing Type"?: string
  Specification?: string
  "Property Type"?: string
}

// Description,Bedrooms / Baths,Type,Price,Size,State,Location,Housing Type,Telephone,Contact Person
export interface RealEstateLead extends BaseLead {
  "Lead Type": "realestate"
  Description: string
  Price: number
  "Bedrooms/Baths": string
  "Contact Person": string
  Type: "Sell" | "Rent"
  Size?: number
  State?: string
  Location?: number
  "Housing Type"?: string
  Specification?: string
  "Property Type"?: string
}

export type Lead = BaseLead | RealEstateLead
export type NewLead = NewBaseLead | NewRealEstateLead
