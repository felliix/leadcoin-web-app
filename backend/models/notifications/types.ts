export interface BaseNotification {
  msg: string
  userId: number
}

export interface NewNotification extends BaseNotification {
  id?: number
  unread: true
}

export interface Notification extends BaseNotification {
  id: number
  unread: boolean
}

export interface NotificationQuery {
  unread: boolean
}
