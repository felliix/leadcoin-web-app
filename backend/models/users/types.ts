type timestamp = number

export interface BaseUserInterface {
  fname: string
  lname: string
  email: string
  disabled?: disabledReason
  emailConfirmationKey?: string
  country?: string
  phone?: string
  company?: string
  balance?: number
}
export interface ExistingUserInterface extends BaseUserInterface {
  password: string
  id: number
  emailConfirmationKey: string
  disabled: disabledReason
  created: number
  balance: number
}
export interface ExistingUserInterfaceCondition {
  id?: number
  fname?: string
  disabled?: disabledReason
  emailConfirmationKey?: string
  lname?: string
  email?: string
  password?: string
  /**? Timestamp */
  created?: number
  provider_id?
  provider?
  country?: string
  phone?: string
  company?: string
}

export interface NewUserInterface extends BaseUserInterface {
  plainPassword?: string
}

export enum disabledReason {
  // do not change strings. Frontend & API depends on it
  EMAIL_NOT_VERIFIED = "EMAIL_NOT_VERIFIED",
  PROFILE_NOT_COMPLETED = "PROFILE_NOT_COMPLETED",
}
