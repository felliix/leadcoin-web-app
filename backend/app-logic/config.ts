var dotenv = require("dotenv")
const url = require("url")
var utils = require("../utils")

dotenv.config()
utils.throwOnMissingEnvironmentVariables()

class Config {
  // FLAGS
  AUTO_CONFIRM_EMAIL = true
  SKIP_COMPLETE_PROFILE = true
  INITIAL_BALANCE = 1000

  env = process.env.NODE_ENV
  backend = process.env.BACKEND
  frontend = process.env.FRONTEND
  upload = process.env.UPLOAD_DIR
  app = {
    port: process.env.PORT,
  }
  auth = {
    jwt: {
      secret: process.env.JWT_SECRET,
      expiresIn: process.env.JWT_EXPIRES_IN,
    },
    google: {
      clientID: process.env.GOOGLE_CLIENT_ID,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
      callbackURL: process.env.BACKEND + "/auth/google/callback",
    },
    linkedin: {
      clientID: process.env.LINKEDIN_CLIENT_ID,
      clientSecret: process.env.LINKEDIN_CLIENT_SECRET,
      callbackURL: process.env.BACKEND + "/auth/linkedin/callback",
    },
  }
  mail = {
    mailer: process.env.MAILER,
    host: process.env.MAIL_HOST,
    port: process.env.MAIL_PORT,
    from: process.env.MAIL_FROM,
    user: process.env.MAIL_USER,
    pass: process.env.MAIL_PASS,
  }
  mysql = {
    host: process.env.MYSQL_HOST,
    port: process.env.MYSQL_PORT,
    user: process.env.MYSQL_USER,
    database: "leadcoin",
    connectionLimit: process.env.CONNECTION_LIMIT || 10,
    password: process.env.MYSQL_PASSWORD,
  }
}

var config = new Config()
export { Config as IConfig }
export default config
