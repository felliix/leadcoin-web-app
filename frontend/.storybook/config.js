import { configure } from "@storybook/react"
import "../src/styles/global.scss"
import { configure as enconf } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

const req = require.context("../src/", true, /.stories.js/)

enconf({ adapter: new Adapter() })

function loadStories() {
  req.keys().forEach(filename => req(filename))
}

configure(loadStories, module)
