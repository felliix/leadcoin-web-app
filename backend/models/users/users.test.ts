import Users from "./users"
import * as Chance from "chance"
import { NewUserInterface } from "./types"
import config from "../../app-logic/config"
var chance = Chance()

import SQL from "../mysql-pool/mysql-pool"
var users = new Users(new SQL(config))

beforeEach(async () => {
  await users.deleteAll()
})

test("createUser", async () => {
  var user_id = await users.createUser({
    disabled: null,
    email: chance.email(),
    fname: "first name",
    lname: "last name",
    plainPassword: "my plain password",
  })
  expect(typeof user_id).toEqual("number")
})

test("getOne (remove created, id from snapshot)", async () => {
  var user_id = await users.createUser({
    disabled: null,
    email: "example_email@example_email.com",
    fname: "first name",
    lname: "last name",
    plainPassword: "my plain password",
  })
  var user = await users.tryGetById(user_id)
  if (user instanceof NotFound) {
    throw new Error("user should be found")
  }
  delete user.password
  delete user.created
  delete user.id
  expect(user).toMatchSnapshot()
})
// test.skip("create a user with missing requried parameter", done => {
//   var p = users.createUser(<NewUserInterface>{})
//   p.catch(() => {
//     done()
//   })
//   p.then(() => {
//     done(new Error("should not be resolved"))
//   })
// })

test("getUserByEmailAndPassword (remove id,created,password from snapshot)", async () => {
  await users.createUser({
    email: "myemail5@myemail.com",
    plainPassword: "a12345",
    disabled: null,
    fname: "",
    lname: "",
  })
  var user = await users.getUserByEmailAndPassword(
    "myemail5@myemail.com",
    "a12345",
  )
  expect(user).not.toBeInstanceOf(NotFound)
  if (!(user instanceof NotFound)) {
    delete user.password
    delete user.created
    delete user.id
  }
  expect(user).toMatchSnapshot()
})

import NotFound from "../../utils/not-found"

test("increaseBalance and decreaseBalance should work", async () => {
  var user_id = await users.createUser({
    email: "myemail@myemail.com",
    plainPassword: "a12345",
    disabled: null,
    fname: "",
    lname: "",
    balance: 0,
  })

  await users.increaseBalance(user_id, 20)
  var t1 = await users.getUserByEmailAndPassword(
    "myemail@myemail.com",
    "a12345",
  )
  expect(t1.balance).toBe(20)
  await users.decreaseBalance(user_id, 10)
  var t1 = await users.getUserByEmailAndPassword(
    "myemail@myemail.com",
    "a12345",
  )
  expect(t1.balance).toBe(10)
})

test("setNewPassword, and not found the user (remove password,created,id from snapshot)", async () => {
  var user_id = await users.createUser({
    email: "myemail@myemail.com",
    plainPassword: "a12345",
    disabled: null,
    fname: "",
    lname: "",
  })

  await users.setNewPassword(user_id, "new-a123")

  var t1 = await users.getUserByEmailAndPassword(
    "myemail@myemail.com",
    "a12345",
  )
  expect(t1).toBeInstanceOf(NotFound)
})

test("setNewPassword, and getUserByEmailAndPassword (remove password,created,id from snapshot)", async () => {
  var user_id = await users.createUser({
    email: "myemail22@myemail.com",
    plainPassword: "a12345",
    disabled: null,
    fname: "",
    lname: "",
  })

  await users.setNewPassword(user_id, "new-a123")
  var t1 = await users.getUserByEmailAndPassword(
    "myemail22@myemail.com",
    // "a12345",
    "new-a123",
  )

  if (t1 instanceof NotFound) {
    throw new Error("should found the user")
  } else {
    delete t1.password
    delete t1.created
    delete t1.id
    expect(t1).toMatchSnapshot()
  }
})
