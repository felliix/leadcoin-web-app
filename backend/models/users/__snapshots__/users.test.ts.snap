// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOne (remove created, id from snapshot) 1`] = `
Object {
  "disabled": null,
  "email": "example_email@example_email.com",
  "fname": "first name",
  "lname": "last name",
}
`;

exports[`getUserByEmailAndPassword (remove id,created,password from snapshot) 1`] = `
Object {
  "disabled": null,
  "email": "myemail5@myemail.com",
  "fname": "",
  "lname": "",
}
`;

exports[`setNewPassword, and getUserByEmailAndPassword (remove password,created,id from snapshot) 1`] = `
Object {
  "disabled": null,
  "email": "myemail22@myemail.com",
  "fname": "",
  "lname": "",
}
`;
